packages/swapper/src/swappers/thorchain/buildThorTrade/buildThorTrade.ts

```diff
@@ -38,20 +48,26 @@ export const buildTrade = async ({ deps, input }: BuildTradeArgs): Promise<ThorT
       sendMax,
     } = input
 
-    const quote = await getThorTradeQuote({ deps, input })
+    const { chainNamespace } = fromAssetId(sellAsset.assetId)
     const sellAdapter = deps.adapterManager.get(sellAsset.chainId)
 
     if (!sellAdapter)
-      throw new SwapError('[buildTrade]: unsupported sell asset', {
-        code: SwapErrorType.BUILD_TRADE_FAILED,
-        fn: 'buildTrade',
-        details: { sellAsset },
-      })
+      return Err(
+        makeSwapErrorRight({
+          message: '[buildTrade]: unsupported sell asset',
+          code: SwapErrorType.BUILD_TRADE_FAILED,
+          details: { sellAsset },
+        }),
+      )
 
-    const { chainNamespace } = fromAssetId(sellAsset.assetId)
+    const maybeQuote = await getThorTradeQuote({ deps, input })
+
+    if (maybeQuote.isErr()) return Err(maybeQuote.unwrapErr())
```

Comment:
This, as long as the if/else logic below, should sit inside a `match` to be more idiomatic.

Reply:
See this comment and conversation: 917175dee5a35b68e4345dbddd71a4f983c43b27

Updated all `unwrapErr()` / `unwrap()` instances that aren't async to `.map()` in 917175dee5a35b68e4345dbddd71a4f983c43b27.
Let's do another pass in a follow-up and iterate on how we can achieve the same for async flows? That's not going to work in such a straightforward with async flows way due to the way promises are boxed into a `Result`, unfortunately.